Cli multisite

timotei-litespeed opened the pull request:
Fixed cli command:
`wp --url=http://test.com/url litespeed-purge url`

When using Multisite and behind LiteSpeed Web ADC

hi-hai commented on cli/purge.cls.php:
```diff
@@ -157,6 +157,7 @@ public function url($args)
 	{
 		$data = array(
 			Router::ACTION => Core::ACTION_QS_PURGE,
+			Router::VALIDATE_PURGE => preg_replace('/.{1,3}(.?)/', '$1', NONCE_KEY),
```
What did this reg do? Why using this way? Could you please give some comments for code readability?

hi-hai commented:
Have you tested your change?

timotei-litespeed commented:
Made changes to the validation:
- tested other actions from CLI and they were failing because of the parameter. Added test to skip validation for Core::ACTION_QS_PURGE
- changed the hash to use Router::get_hash() (saw it in the code and I think it fits better)

timotei-litespeed commented:
@hi-hai please review the last push "[Router - Get hash improvement](https://github.com/litespeedtech/lscache_wp/pull/690/commits/29c0134c8b92c89c7f049b839be9fa475a0dcb19)" 
Scope of the change is to allow new hashes to be generated and easy to use in future needs

hi-hai commented on src/router.cls.php:
```diff
@@ -501,12 +506,30 @@ private function verify_action()
 			}
 
+			$save_blog = get_current_blog_id();
+			if ($_REQUEST['switch_blog']) {
```
Key may not exist, which will result in warning. Please use `!empty()` to check

timotei-litespeed replied:
done

timotei-litespeed replied:
This was replaced with new code.

hi-hai commented on src/router.cls.php:
```diff
@@ -501,12 +506,30 @@ private function verify_action()
+			$hash = Router::get_hash(self::VALIDATE_PURGE);
+			if ($_REQUEST['switch_blog']) {
+				// Restore blog if needed.
+				switch_to_blog($save_blog);
```
This switch back and forth is a bit complicated. Should store the hash in the target blog. Please switch at https://github.com/litespeedtech/lscache_wp/pull/690/files#diff-7efe974c01bd97f4667ac581f8ce7fdf45b700734b28b2c945059af3e4b4059aR160

hi-hai commented on cli/purge.cls.php:
```diff
@@ -166,7 +166,30 @@ public function url($args)
+			$path = '/';
+			// If subfolder install test if we can identify the blog from url.
+			// If subdomain install: $path needs to remain '/'.
+			if(!SUBDOMAIN_INSTALL){
```
Directly calling a const is not safe. Please use `is_subdomain_install()`.
If you want to use const, use like how its used in `is_subdomain_install()` defination.

timotei-litespeed replied:
Changed

timotei-litespeed commented on cli/purge.cls.php:
```diff
@@ -166,7 +166,30 @@ public function url($args)
+			}
+
+			// Test if link can be found.
+			if (get_blog_id_from_url($deconstructed['host'], $path) === 0) {
```
This is used for test if subdomain install has the domain url sent

hi-hai commented on src/router.cls.php:
```diff
@@ -501,12 +519,20 @@ private function verify_action()
 				return;
 			}
 
+			$hash = Router::get_hash(self::VALIDATE_PURGE);
```
This should be under the cond `$action == Core::ACTION_QS_PURGE`, otherwise it will update hash value for all verify_action()
